packages/components/src/button/deprecated.tsx

```diff
@@ -8,9 +12,23 @@ import { forwardRef } from '@wordpress/element';
 /**
  * Internal dependencies
  */
-import Button from '../button';
+import Button from '.';
+import type { ButtonProps, DeprecatedIconButtonProps, TagName } from './types';
+import type { WordPressComponentProps } from '../ui/context';
 
-function IconButton( { labelPosition, size, tooltip, label, ...props }, ref ) {
+function UnforwardedIconButton(
+	{
+		label,
+		labelPosition,
+		size,
+		tooltip,
+		...props
+	}: WordPressComponentProps<
+		ButtonProps & DeprecatedIconButtonProps,
+		TagName
+	>,
```

Comment:
How about we go a little more DRY here:

```suggestion
	}: React.ComponentPropsWithoutRef< typeof Button > &
		DeprecatedIconButtonProps,
```

Reply:
Good idea! Committed in https://github.com/WordPress/gutenberg/pull/46997/commits/3632a42bc8eef7f6167a24bc2d2330881f35ad5c